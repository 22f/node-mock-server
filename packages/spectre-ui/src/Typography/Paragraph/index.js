// @flow
/* global props defaultProps */

import "spectre.css";
import React from "react";
import type { ParagraphPropsType } from "../../../spectre-ui.js.flow";

const Paragraph = ({ children, innerHtml, lang, className }: ParagraphPropsType): React$Element<*> => {
	if (typeof innerHtml === "string") {
		let langAttr = "";
		if (lang) {
			langAttr = ` lang="${lang}"`;
		}
		return <div className={className} dangerouslySetInnerHTML={{ __html: `<p${langAttr}>${innerHtml}</p>` }} />;
	}
	return (
		<p className={className} lang={lang}>
			{children}
		</p>
	);
};

Paragraph.displayName = "Typography/Paragraph";

export default Paragraph;
