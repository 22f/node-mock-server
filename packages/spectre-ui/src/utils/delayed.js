// @flow

const delayed = (fn: Function, delay?: number = 1): string => {
	const timerId: string = window.setTimeout((): Function => {
		window.clearTimeout(timerId);
		return fn.call();
	}, delay);

	return timerId;
};

export default delayed;
