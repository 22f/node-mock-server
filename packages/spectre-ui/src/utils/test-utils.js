/* global describe it expect */
import React from "react";
import { shallow } from "enzyme";

export function itWithoutProp(Component, props, propName: string) {
	it(`snapshot: no ${propName}`, () => {
		const _props = { ...props, [propName]: undefined };
		const tree = shallow(<Component {..._props} />);
		expect(tree).toMatchSnapshot();
	});
}
export function itOptionalProps(Component, props, propNames: Array<string>) {
	propNames.forEach((propName: string) => {
		itWithoutProp(Component, props, propName);
	});
}
export function itEmptyArrayProp(Component, props, propName: string) {
	it(`snapshot: empty ${propName}`, () => {
		const _props = { ...props, [propName]: [] };
		const tree = shallow(<Component {..._props} />);
		expect(tree).toMatchSnapshot();
	});
}
export function itArrayProps(Component, props, propNames: Array<string>) {
	propNames.forEach((propName: string) => {
		itEmptyArrayProp(Component, props, propName);
	});
}
