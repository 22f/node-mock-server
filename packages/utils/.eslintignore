/.idea/
/node_modules/
/dist/
/coverage/
/flow-typed/
/node-mock-server-utils.js.flow
