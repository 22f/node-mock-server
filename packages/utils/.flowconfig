[ignore]
<PROJECT_ROOT>/node_modules/.*.json
<PROJECT_ROOT>/dist/.*
.*/log/.*

[include]

[libs]
./node_modules/node-mock-server-log/node-mock-server-log.js.flow

[options]
unsafe.enable_getters_and_setters=true
esproposal.decorators=ignore
