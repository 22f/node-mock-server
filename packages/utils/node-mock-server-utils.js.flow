export type TypeOfValueType = null | string | number | Object | Array<any>;
export type TypeOfType = 'undefined' | 'null' | 'string' | 'symbol' | 'Array' | 'Function' | 'Object';
export type IsType = {
	undefined: (value: any) => boolean,
	null: (value: any) => boolean,
	string: (value: any) => boolean,
	number: (value: any) => boolean,
	boolean: (value: any) => boolean,
	symbol: (value: any) => boolean,
	array: (value: any) => boolean,
	function: (value: any) => boolean,
	buffer: (value: any) => boolean,
	object: (value: any) => boolean,
	regExp: (value: any) => boolean,
	date: (value: any) => boolean,
	error: (value: any) => boolean,
	nativePromise: (value: any) => boolean,
	promise: (value: any) => boolean,
	generator: (value: any) => boolean,
	generatorFunction: (value: any) => boolean,
	map: (value: any) => boolean,
	set: (value: any) => boolean,
	weakMap: (value: any) => boolean,
	weakSet: (value: any) => boolean,
	nan: (value: any) => boolean,
	nullOrUndefined: (value: any) => boolean,
	primitive: (value: any) => boolean,
	integer: (value: any) => boolean,
	plainObject: (value: any) => boolean,
	inRange: (value: any) => boolean,
	domElement: (value: any) => boolean,
	empty: (value: any) => boolean,
};
export type MethodEnumType = 'GET' | 'PUT' | 'POST' | 'OPTIONS' | 'DELETE';

declare module 'node-mock-server-utils' {

	declare export type $MethodEnumType = MethodEnumType;

	declare export function readDir(dirPath: string, ignore: ?Array<string | RegExp>): Array<string>;
	declare export function writeDir(dirPath: string): void;
	declare export function removeDir(dirPath: string): boolean;
	declare export function isDir(dirPath: string): boolean;

	declare export function readFile(filePath: string): ?string;
	declare export function writeFile(filePath: string, data: string): void;
	declare export function removeFile(filePath: string): boolean;
	declare export function isFile(dirPath: string): boolean;

	declare export function capitalize(value: string): string;
	declare export function toCamelCase(value: string): string;
	declare export function toPascalCase(value: string): string;

	declare export function typeOf(value: TypeOfValueType): TypeOfType;
	declare export var is: IsType;

	declare export function toMethodEnum(value: string): $MethodEnumType;
}
