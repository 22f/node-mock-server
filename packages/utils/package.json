{
    "name": "node-mock-server-utils",
    "version": "0.1.0",
    "private": false,
    "dependencies": {
        "@sindresorhus/is": "0.3.0",
        "camel-case": "3.0.0",
        "capitalize": "1.0.0",
        "del": "3.0.0",
        "make-dir": "1.0.0",
        "node-mock-server-log": "0.1.0",
        "to-pascal-case": "1.0.0"
    },
    "devDependencies": {
        "eslint-config": "0.1.0",
        "flow-bin": "0.61.0",
        "flow-typed": "2.2.3",
        "node-flow-config": "0.1.0"
    },
    "main": "dist/index.js",
    "scripts": {
        "dev": "node-flow-config start",
        "build": "node-flow-config build",
        "test": "CI=true node-flow-config test --env=jsdom --coverage",
        "test:dev": "node-flow-config test --env=jsdom",
        "lint": "npm-run-all lint:*",
        "lint:js": "node_modules/.bin/eslint .",
        "flow": "node_modules/.bin/flow",
        "flow-restart": "node_modules/.bin/flow stop && node_modules/.bin/flow",
        "flow-install": "node_modules/.bin/flow-typed install -i dev"
    }
}
