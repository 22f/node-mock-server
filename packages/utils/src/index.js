// @flow

// readDir
export * from './read-dir';
// writeDir -> https://www.npmjs.com/package/make-dir
export * from './write-dir';
// removeDir -> https://github.com/sindresorhus/del
export * from './remove-dir';
// isDir
export * from './is-dir';

// readFile
export * from './read-file';
// writeFile
export * from './write-file';
// removeFile
export * from './remove-file';
// isFile
export * from './is-file';

// capitalize -> https://www.npmjs.com/package/capitalize
export * from './capitalize';
// toCamelCase -> https://www.npmjs.com/package/camel-case
export * from './to-camel-case';
// toPascalCase -> https://www.npmjs.com/package/to-pascal-case
export * from './to-pascal-case';

// is -> https://www.npmjs.com/package/@sindresorhus/is
export * from './is';

export * from './method-enum';
