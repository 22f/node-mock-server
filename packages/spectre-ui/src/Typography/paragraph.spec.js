/* global describe it expect */

import React from "react";
import { shallow } from "enzyme";
import { itOptionalProps } from "../utils/test-utils";
import Paragraph from "./paragraph";

const props: Object = {
	children: <div />,
	innerHtml: "<span>Inner Html</span>",
	lang: "zh-Hans",
	className: "class-name"
};

describe("Blockquote", () => {
	describe("render", () => {
		it("snapshot: default", () => {
			const _props = { ...props };
			const tree = shallow(<Paragraph {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		itOptionalProps(Paragraph, props, ["className", "children", "innerHtml", "lang"]);
	});
});
