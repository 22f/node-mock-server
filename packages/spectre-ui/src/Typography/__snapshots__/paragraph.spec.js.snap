// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote render snapshot: default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="class-name"
    dangerouslySetInnerHTML={
        Object {
            "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
          }
    }
/>,
  "nodes": Array [
    <div
      className="class-name"
      dangerouslySetInnerHTML={
            Object {
                  "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Paragraph
        className="class-name"
        innerHtml="<span>Inner Html</span>"
        lang="zh-Hans"
>
        <div />
</Paragraph>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Paragraph {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "className": "class-name",
          "innerHtml": "<span>Inner Html</span>",
          "lang": "zh-Hans",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Paragraph
    className="class-name"
    innerHtml="<span>Inner Html</span>"
    lang="zh-Hans"
>
    <div />
</Paragraph>,
}
`;

exports[`Blockquote render snapshot: no children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="class-name"
    dangerouslySetInnerHTML={
        Object {
            "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
          }
    }
/>,
  "nodes": Array [
    <div
      className="class-name"
      dangerouslySetInnerHTML={
            Object {
                  "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Paragraph
        className="class-name"
        innerHtml="<span>Inner Html</span>"
        lang="zh-Hans"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Paragraph {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": undefined,
          "className": "class-name",
          "innerHtml": "<span>Inner Html</span>",
          "lang": "zh-Hans",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Paragraph
    className="class-name"
    innerHtml="<span>Inner Html</span>"
    lang="zh-Hans"
/>,
}
`;

exports[`Blockquote render snapshot: no className 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className=""
    dangerouslySetInnerHTML={
        Object {
            "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
          }
    }
/>,
  "nodes": Array [
    <div
      className=""
      dangerouslySetInnerHTML={
            Object {
                  "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Paragraph
        className={undefined}
        innerHtml="<span>Inner Html</span>"
        lang="zh-Hans"
>
        <div />
</Paragraph>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Paragraph {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "className": undefined,
          "innerHtml": "<span>Inner Html</span>",
          "lang": "zh-Hans",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className=""
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
        "_debugID": 4,
        "_renderedOutput": <div
          className=""
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p lang=\\"zh-Hans\\"><span>Inner Html</span></p>",
                            }
          }
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Paragraph
    className={undefined}
    innerHtml="<span>Inner Html</span>"
    lang="zh-Hans"
>
    <div />
</Paragraph>,
}
`;

exports[`Blockquote render snapshot: no innerHtml 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <p
    className="class-name"
    lang="zh-Hans"
>
    <div />
</p>,
  "nodes": Array [
    <p
      className="class-name"
      lang="zh-Hans"
>
      <div />
</p>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Paragraph
        className="class-name"
        innerHtml={undefined}
        lang="zh-Hans"
>
        <div />
</Paragraph>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Paragraph {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "className": "class-name",
          "innerHtml": undefined,
          "lang": "zh-Hans",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <p
          className="class-name"
          lang="zh-Hans"
>
          <div />
</p>,
        "_debugID": 8,
        "_renderedOutput": <p
          className="class-name"
          lang="zh-Hans"
>
          <div />
</p>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Paragraph
    className="class-name"
    innerHtml={undefined}
    lang="zh-Hans"
>
    <div />
</Paragraph>,
}
`;

exports[`Blockquote render snapshot: no lang 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="class-name"
    dangerouslySetInnerHTML={
        Object {
            "__html": "<p><span>Inner Html</span></p>",
          }
    }
/>,
  "nodes": Array [
    <div
      className="class-name"
      dangerouslySetInnerHTML={
            Object {
                  "__html": "<p><span>Inner Html</span></p>",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Paragraph
        className="class-name"
        innerHtml="<span>Inner Html</span>"
        lang={undefined}
>
        <div />
</Paragraph>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Paragraph {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "className": "class-name",
          "innerHtml": "<span>Inner Html</span>",
          "lang": undefined,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p><span>Inner Html</span></p>",
                            }
          }
/>,
        "_debugID": 10,
        "_renderedOutput": <div
          className="class-name"
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<p><span>Inner Html</span></p>",
                            }
          }
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Paragraph
    className="class-name"
    innerHtml="<span>Inner Html</span>"
    lang={undefined}
>
    <div />
</Paragraph>,
}
`;
